pkg/image/admission/admission.go

```diff
@@ -0,0 +1,299 @@
+	}
+	plugin.limitRanger = limitRanger
+
+	// NOTE: the reflector is set up in SetOpenshiftClient to ensure we have access to the client
+	return plugin, nil
+}
+
+// Run starts the reflectors used by the plugin.
+func (a *imageLimitRangerPlugin) Run() {
+	if a.stopChan == nil {
+		a.stopChan = make(chan struct{})
+		a.imageStreamReflector.RunUntil(a.stopChan)
+	}
+}
+
+// Stop stops the reflectors used by the plugin.
+func (a *imageLimitRangerPlugin) Stop() {
+	if a.stopChan != nil {
+		close(a.stopChan)
+		a.stopChan = nil
+	}
+}
+
+// Admit invokes the admission logic for checking against LimitRanges.
+func (a *imageLimitRangerPlugin) Admit(attr kadmission.Attributes) error {
+	if (attr.GetResource() != imageapi.Resource("imagestreammappings") &&
+		attr.GetResource() != imageapi.Resource("imagestreamtags") &&
+		attr.GetResource() != imageapi.Resource("imagestreams")) ||
+		attr.GetSubresource() != "" {
+
+		return nil // not applicable
+	}
+
+	// If the namespace is not set on the object ensure it gets set.
+	if om, err := kapi.ObjectMetaFor(attr.GetObject()); err == nil {
+		if om.Namespace == "" {
+			glog.V(5).Infof("%s: namespace was not set for object, defaulting to %s", pluginName, attr.GetNamespace())
+			om.Namespace = attr.GetNamespace()
+		}
+	}
+
+	return a.limitRanger.Admit(attr)
+}
+
+// SetOpenshiftClient provides init functionality for anything that requires the OpenShift client.
+func (a *imageLimitRangerPlugin) SetOpenshiftClient(c client.Interface) {
+	a.osClient = c
+
+	reflector := cache.NewReflector(
+		&cache.ListWatch{
+			ListFunc: func(options kapi.ListOptions) (runtime.Object, error) {
+				return a.osClient.ImageStreams(kapi.NamespaceAll).List(options)
+			},
+			WatchFunc: func(options kapi.ListOptions) (watch.Interface, error) {
+				return a.osClient.ImageStreams(kapi.NamespaceAll).Watch(options)
+			},
+		},
+		&imageapi.ImageStream{},
+		a.imageStreamStore,
+		0,
+	)
+	a.imageStreamReflector = reflector
+
+	a.Run()
+}
+
+func (a *imageLimitRangerPlugin) Validate() error {
+	if a.osClient == nil {
+		return fmt.Errorf("%s needs an Openshift client", pluginName)
+	}
+	return nil
+}
+
+// imageLimitRanger is the limit range implementation that checks resource against the
+// image limit ranges.
+func (a *imageLimitRangerPlugin) imageLimitRanger(limitRange *kapi.LimitRange, kind string, obj runtime.Object) error {
+	if !limitAppliesToImages(limitRange) {
+		return nil
+	}
+
+	var images []imageapi.Image
+
+	switch obj.(type) {
+	case *imageapi.ImageStreamMapping:
+		ism, ok := obj.(*imageapi.ImageStreamMapping)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStreamMapping: %#v", pluginName, obj)
+			return nil
+		}
+		images = []imageapi.Image{ism.Image}
+	case *imageapi.ImageStreamTag:
+		tag, ok := obj.(*imageapi.ImageStreamTag)
+		if !ok {
+			glog.V(4).Infof("%s: unable to convert obj to ImageStreamTag: %#v", pluginName, obj)
+			return nil
+		}
+		images = []imageapi.Image{tag.Image}
```

Comment:
So thanks to the additional extended test I wrote today. I [found out](https://github.com/openshift/origin/pull/7930/commits/d5ea1b618032b8b4c91ce4cc1309bff93599401b#r56545173) this is incorrect.

We need to check `tag.Tag` instead.